src/libs/bitmap/src/lib.rs

```diff
@@ -170,40 +157,61 @@ impl Bitmap {
     /// instead.
     ///
     pub fn alloc_range(&mut self, size: usize) -> Result<usize, Error> {
-        // Check if the size is invalid.
-        if (size == 0) || (size > u8::BITS as usize) {
+        // Check if the size is valid.
+        if size == 0 || size > self.number_of_bits {
             let reason: &str = "invalid size";
             return Err(Error::new(ErrorCode::InvalidArgument, reason));
         }
 
-        // Check if the size is out of bounds.
-        if size > self.number_of_bits {
-            let reason: &str = "size out of bounds";
-            return Err(Error::new(ErrorCode::InvalidArgument, reason));
+        // Check if allocation exceeds the bitmap capacity.
+        if self.usage > self.number_of_bits - size {
+            let reason: &str = "allocation exceeds bitmap capacity";
+            return Err(Error::new(ErrorCode::OutOfMemory, reason));
         }
 
-        // Compute mask.
-        let mask: u8 = if size == u8::BITS as usize {
-            // Use u8::MAX directly to avoid overflow
-            u8::MAX
-        } else {
-            ((1 << size) - 1) as u8
-        };
-
-        // Traverse the bitmap one word at a time.
-        for (i, word) in self.bits.iter_mut().enumerate() {
-            // Check if this word is not full.
-            if *word != u8::MAX {
-                // Find a free range of bits.
-                for j in 0..=(u8::BITS as usize - size) {
-                    // Check if the range is free.
-                    if (*word & (mask << j)) == 0 {
-                        // It is, thus allocate it.
-                        *word |= mask << j;
-                        self.usage += size;
-                        return Ok(i * u8::BITS as usize + j);
-                    }
+        debug_assert_eq!(
+            self.bits.len() * u8::BITS as usize,
+            self.number_of_bits,
+            "bitmap length must match the number of bits"
+        );
+
+        let mut start: usize = 0;
+
+        // Traverse the bitmap until the last possible starting bit.
+        while start <= self.number_of_bits - size {
+            // Check for fast skip/ path.
+            let word: usize = start / u8::BITS as usize;
+            let is_aligned: bool = start % u8::BITS as usize == 0;
+            if is_aligned {
+                // Fast skip: if the starting word is full, skip to the next word.
+                if self.bits[word] == u8::MAX {
+                    // Jump to next byte boundary.
+                    let next_start: usize = (word + 1) * u8::BITS as usize;
```

Comment:
NIT, wouldn't be simpler to just do start + u8::BITS instead of the multiplication, it makes it more explicit that you are moving one word ahead.